homeassistant/components/lg_thinq/vacuum.py

```diff
@@ -0,0 +1,185 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+DEVICE_TYPE_VACUUM_MAP: dict[DeviceType, tuple[StateVacuumEntityDescription, ...]] = {
+    DeviceType.ROBOT_CLEANER: (
+        StateVacuumEntityDescription(
+            key=ExtendedProperty.VACUUM,
+            name=None,
+        ),
+    ),
+}
+
+
+class State(StrEnum):
+    """State of device."""
+
+    HOMING = "homing"
+    PAUSE = "pause"
+    RESUME = "resume"
+    SLEEP = "sleep"
+    START = "start"
+    WAKE_UP = "wake_up"
+
+
+ROBOT_STATUS_TO_HA = {
+    "charging": STATE_DOCKED,
+    "diagnosis": STATE_IDLE,
+    "homing": STATE_RETURNING,
+    "initializing": STATE_IDLE,
+    "macrosector": STATE_IDLE,
+    "monitoring_detecting": STATE_IDLE,
+    "monitoring_moving": STATE_IDLE,
+    "monitoring_positioning": STATE_IDLE,
+    "pause": STATE_PAUSED,
+    "reservation": STATE_IDLE,
+    "setdate": STATE_IDLE,
+    "sleep": STATE_IDLE,
+    "standby": STATE_IDLE,
+    "working": STATE_CLEANING,
+    "error": STATE_ERROR,
+}
+ROBOT_BATT_TO_HA = {
+    "moveless": 5,
+    "dock_level": 5,
+    "low": 30,
+    "mid": 50,
+    "high": 90,
+    "full": 100,
+    "over_charge": 100,
+}
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an entry for vacuum platform."""
+    entities: list[ThinQStateVacuumEntity] = []
+    for coordinator in entry.runtime_data.coordinators.values():
+        if (
+            descriptions := DEVICE_TYPE_VACUUM_MAP.get(
+                coordinator.api.device.device_type
+            )
+        ) is not None:
+            for description in descriptions:
+                entities.extend(
+                    ThinQStateVacuumEntity(coordinator, description, property_id)
+                    for property_id in coordinator.api.get_active_idx(description.key)
+                )
+
+    if entities:
+        async_add_entities(entities)
+
+
+class ThinQStateVacuumEntity(ThinQEntity, StateVacuumEntity):
+    """Represent an thinq vacuum platform."""
+
+    def __init__(
+        self,
+        coordinator: DeviceDataUpdateCoordinator,
+        entity_description: StateVacuumEntityDescription,
+        property_id: str,
+    ) -> None:
+        """Initialize vacuum platform."""
+        super().__init__(coordinator, entity_description, property_id)
+
+        self._attr_supported_features = (
+            VacuumEntityFeature.SEND_COMMAND
+            | VacuumEntityFeature.STATE
+            | VacuumEntityFeature.BATTERY
+            | VacuumEntityFeature.START
+            | VacuumEntityFeature.PAUSE
+            | VacuumEntityFeature.RETURN_HOME
+        )
+        self._device_state: str | None = None
```

Comment:
This one as well. That way you can remove the constrcutor

Reply:
You're right. I removed the constructor.

Reply:
Moved.
But we need to declare variables in __init__ as below.
Otherwise, it generates error in pylint.

homeassistant/components/lg_thinq/vacuum.py:116:8: W0201: Attribute '_device_state' defined outside __init__ (attribute-defined-outside-init)